godel_surface_detection/include/godel_surface_detection/data_coordinator/data_coordinator.h

```diff
@@ -0,0 +1,391 @@
+
+
+    /**
+     * @brief Set the mesh of the specified record
+     * @param id ID of the desired record
+     * @param mesh PolygonMesh representing the surface of interest
+     * @return true if record is found, false otherwise
+     */
+    bool setSurfaceMesh(int id, pcl::PolygonMesh mesh)
+    {
+      for(auto& rec : records_)
+      {
+        if(id == rec.id_)
+        {
+          rec.surface_mesh_ = mesh;
+          return true;
+        }
+      }
+
+      ROS_ERROR_STREAM(UNABLE_TO_FIND_RECORD_ERROR << " " << id);
+      return false;
+    }
+
+
+    /**
+     * @brief getSurfaceMesh
+     * @param id ID of the desired record
+     * @param mesh Destination for the mesh
+     */
+    void getSurfaceMesh(int id, pcl::PolygonMesh& mesh)
+    {
+      for(auto& rec : records_)
+      {
+        if(id == rec.id_)
+        {
+          mesh = rec.surface_mesh_;
+          break;
+        }
+      }
+    }
+
+
+    /**
+     * @brief Add poses comprising the edge of a surface to its record
+     * @param id ID of the desired record
+     * @param name The name of the edge
+     * @param edge_poses PoseArray of edges
+     * @return true if record is found and edge is added, false otherwise
+     */
+    bool addEdge(int id, std::string name, geometry_msgs::PoseArray edge_poses)
+    {
+      for(auto& rec : records_)
+      {
+        if(id == rec.id_)
+        {
+          std::pair<std::string, geometry_msgs::PoseArray> edge_pair;
+          edge_pair.first = name;
+          edge_pair.second = edge_poses;
+          rec.edge_pairs_.push_back(edge_pair);
+          return true;
+        }
+      }
+
+      ROS_ERROR_STREAM(UNABLE_TO_FIND_RECORD_ERROR << " " << id);
+      return false;
+    }
+
+
+    /**
+     * @brief renameEdge Change name for an edge
+     * @param id ID of the desired record
+     * @param old_name
+     * @param new_name
+     * @return true if both record is found and an edge with old_name exists within the record, false otherwise
+     */
+    bool renameEdge(int id, std::string old_name, std::string new_name)
+    {
+      for(auto& rec : records_)
+      {
+        if(id == rec.id_)
+        {
+          for(auto& pair: rec.edge_pairs_)
+          {
+            if(old_name.compare(new_name) == 0)
+            {
+              pair.first = new_name;
+              return true;
+            }
+          }
+
+          ROS_WARN_STREAM("Unable to find edge path with name: " << old_name);
+          return false;
+        }
+      }
+
+      ROS_ERROR_STREAM(UNABLE_TO_FIND_RECORD_ERROR << " " << id);
+      return false;
+    }
+
+
+    /**
+     * @brief getEdgePosesByName Obtains PoseArray for edge with name edge_name
+     * @param edge_name
+     * @param edge_poses Destination for PoseArray
+     * @return true if record is found and an edge with edge_name exists within the record, false otherwise
+     */
+    bool getEdgePosesByName(const std::string& edge_name, geometry_msgs::PoseArray& edge_poses)
```

Comment:
Fair point. As written, there is an implicit requirement that edges have unique names (effectively local to the surface right now, but potentially global later). This allows the UI to request visualization of an edge in a single service call with looser coupling to the surface_detection node.

Edge names are implemented as surface names suffixed with "_edge" and an increment (e.g. 1_edge3). We don't currently allow changes to the edge name through the UI, and renameEdges was intended to work in concert with an attempt to rename a surface. However, we could get right back into a situation where our default edge identifier is ambiguous if we expand what we allow the user to do.

Best plan might be to add an id field for edges then pass the field to the UI and let it coordinate the string it presents to the user with the unique id of the edge.